lib/ansible/modules/cloud/amazon/aws_organization_units.py

```diff
@@ -0,0 +1,284 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Kamil Potrec <kamilpotrec@googlemail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = """
+module: aws_organization_units
+short_description: Manages AWS Organizations Units
+description:
+  - Creates and deletes AWS OrganizationalUnits.
+version_added: "2.10"
+author:
+  - Kamil Potrec (@p0tr3c)
+options:
+  name:
+    description:
+      - Name of the organizational unit
+        This can be full OU path such as "Prod/IT/Service_Desk".
+        The parent OUs will be recursively dereferenced.
+        The lookup always starts from the root node, as the names are not unique within the organization tree.
+    required: true
+    type: str
+  state:
+    description:
+      - State of the organizational unit
+    default: present
+    type: str
+    choices:
+      - present
+      - absent
+extends_documentation_fragment:
+  - aws
+  - ec2
+requirements:
+  - boto3
+  - botocore
+"""
+
+EXAMPLES = """
+- name: Create organizational unit
+  aws_organizations:
+    name: Prod
+    state: present
+
+- name: Delete organizational unit
+  aws_organizations:
+    name: Prod
+    state: absent
+"""
+
+RETURN = """
+name:
+  description:
+    - Name of the organizational unit.
+      If the parent OU does not exist the module will fail with warning message.
+  returned: always
+  type: str
+  sample: Prod
+state:
+  description: State of the organization unit.
+  returned: always
+  type: str
+  sample: present
+ou:
+  description: The Organization Unit details.
+  returned: always
+  type: dict
+  sample:
+    {
+        "Arn": "arn:aws:organizations::account:ou/o-id/ou-id",
+        "Id": "ou-id",
+        "Name": "Prod"
+    }
+"""
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError, BotoCoreError
+except ImportError:
+    # handled by AnsibleAWSModule
+    pass
+
+from ansible.module_utils.ec2 import AWSRetry, boto3_tag_list_to_ansible_dict, ansible_dict_to_boto3_tag_list, camel_dict_to_snake_dict
+from ansible.module_utils.aws.core import AnsibleAWSModule, is_boto3_error_code
+from ansible.module_utils._text import to_native
+
+
+class AwsOrganizations():
+    def __init__(self, module):
+        self.module = module
+        try:
```

Comment:
I *believe*  this try/except is redundant and handled within boto3_conn (inside the module.client call) @jillr or @s-hertel may know the code better...